### PR TITLE
Docs: initial user experience + refactor project dependencies

### PR DESCRIPTION
**Pull request by brunobowden**
- Reordered README to sequence guide and put important info higher
- J2ObjC install instructions
- Build commands, e.g. ./gradlew shared:build shared:j2objcXcode
- Explain J2ObjC Standard Libraries, e.g. guava, junit…
- `File: PATH` comments in code to refer to full path, not just name
- App target from “IosApp” => “IOS_APP”
- j2objc => J2ObjC when referred to as a name
- Reworked project dependency FAQs

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - PTAL.

In particular, please check my changed to the project dependencies in the FAQ. I've changed it to be from the perspective of adding a dependency to the `shared` project as that'll be the common use case.

**Comment by confile** [comment]
LGTM

**Comment by brunobowden** [comment]
@advayDev1 - please consider adding an FAQ item on how to use --build-closure.

File: FAQ.md [hunk 1]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
 
-2. Java and j2objc project B depends on a
+In the future, this kind of dependency should be inferred automatically from the corresponding
+Java dependency - [issue 41](https://github.com/j2objc-contrib/j2objc-gradle/issues/41).
+
+
+### How do I setup a dependency on a native library?
+
+1.  Java and J2ObjC project `shared` depends on a
 [custom native library](https://docs.gradle.org/current/userguide/nativeBinaries.html#N15F82)
```
Comment (advayDev1) [inline FAQ.md]: no this section doesn't apply to gradle custom native libraries.  i'd call it prebuilt native library instead

File: FAQ.md [hunk 2]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
 ```
+    // File: shared/build.gradle
     j2objcConfig {
-        extraNativeLib project: ':A', library: 'someLibrary', linkage: 'static'
+        extraNativeLib library: 'lib/someLibrary', linkage: 'static'
     }
 ```
 
-3. Java and j2objc project B depends on library libpreBuilt pre-built outside of
+2.  Java and J2ObjC project `shared` depends on a
```
Comment (advayDev1) [inline FAQ.md]: one of your changes has broken the numbering, checkout the markdown view, they are all 1s

File: FAQ.md [hunk 3]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
+[custom native library](https://docs.gradle.org/current/userguide/nativeBinaries.html#N15F82)
```
Comment (advayDev1) [inline FAQ.md]: break out into its own heading, "How do I setup a dependency on a Gradle native library project"

File: FAQ.md [hunk 4]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
+Project A needs to have the J2objc Gradle Plugin applied and the `j2objcConfig` with the
+`finalConfigure()` call. This applies transitively, so in turn it may need `dependsOnJ2objc`
+again. Alternatively you can try building using `--build-closure` (TODO: need item on this).
 
-2. Java and j2objc project B depends on a
+In the future, this kind of dependency should be inferred automatically from the corresponding
+Java dependency - [issue 41](https://github.com/j2objc-contrib/j2objc-gradle/issues/41).
+
+
+### How do I setup a dependency on a native library?
```
Comment (advayDev1) [inline FAQ.md]: "prebuilt native library"

File: FAQ.md (hunk 1 above)
Comment (advayDev1) [inline FAQ.md]: and remove this link

File: FAQ.md [hunk 5]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
+directory, e.g `A/lib/someLibrary`. Add to `shared/build.gradle`:
+
+```
+    // File: shared/build.gradle
+    j2objcConfig {
+        extraNativeLib project: ':A', library: 'lib/someLibrary', linkage: 'static'
+    }
+```
+
+3.  Java and J2ObjC project `shared` depends on library libpreBuilt pre-built outside of
```
Comment (advayDev1) [inline FAQ.md]: Combine this with the prebuilt subsection #1

File: FAQ.md [hunk 6]
```diff
@@ -106,60 +106,86 @@ See: [How do I enable ARC for my Objective-C classes?](#how-do-i-enable-arc-for-
...
+again. Alternatively you can try building using `--build-closure` (TODO: need item on this).
 
-2. Java and j2objc project B depends on a
+In the future, this kind of dependency should be inferred automatically from the corresponding
+Java dependency - [issue 41](https://github.com/j2objc-contrib/j2objc-gradle/issues/41).
+
+
+### How do I setup a dependency on a native library?
+
+1.  Java and J2ObjC project `shared` depends on a
```
Comment (advayDev1) [inline FAQ.md]: this new subsection you added will not work.  the library: syntax can be used only for libraries built in gradle.

in order to link to a prebuilt .a file, you use:
https://github.com/j2objc-contrib/j2objc-gradle/pull/328/files#diff-e6f81a9d8df4fb4d787121a242ea6bf1R167

**Comment by advayDev1** [comment]
please remove the subsection about linking to a library in shared/lib/something.a - it won't work.
will take a look after you rearrange after reverting that part

File: FAQ.md (hunk 2 above)
Comment (brunobowden) [inline FAQ.md]: Markdown is flaky... if you indent the quote-quote-quote part to start and end the code section then it'll work.

File: FAQ.md (hunk 4 above)
Comment (brunobowden) [inline FAQ.md]: done

**Comment by brunobowden** [comment]
Updated the commit. Please take another look. I'd like to get this squared away before we talk at noon.

**Comment by advayDev1** [comment]
discussed/revised over VC.  lgtm

**Pull request merged**